### DISCUSSION TITLE
Command line parameters

### DISCUSSION THREAD

**Discussion by acataluddi** [iterate-ch/cyberduck#14313] [Q&A]
Hi,

I was wondering whether it was possible to launch CyberDuck using command-line parameters in order to let the application connect to a remote host. For example:

```bash
cyberduck --user root --host host1.example.com --protocol sftp
```

I'm aware of the `duck` command but my ask is to have the CyberDuck UI open and implement an integration between other tools.

Thanks,  
Adriano

**Comment by dkocher** [comment]
You can pass a `.cyberduckbookmark` file to connect to the server like 

```
/Applications/Cyberduck.app/Contents/MacOS/Cyberduck ~/Library/Group\ Containers/G69SCX94XU.duck/Library/Application\ Support/duck/Bookmarks/50d3775f-fe75-477c-85e7-7cc801541f76.duck
```

**Comment by acataluddi** [comment]
Thank you @dkocher,

That would be perfect! I've just tried the command (by the way, I've refactored it so to point to an existing .duck file on my MacBook). However, CyberDuck opens but does not connect to it.

Is it possibile I should add anything more like `--connect`to the command line?

I have installed version 8.5.6 (39394)

Thanks,
Adriano

**Comment by acataluddi** [comment]
Hi David,

I looked at the source code, and unless I'm missing something else, I think the `.duck` file loading is not implemented in macOS. In fact, looking at the entry point function [here](https://github.com/iterate-ch/cyberduck/blob/e61b860a13d4b4438a9c8fba82a61cb6efbbdae6/osx/src/main/java/ch/cyberduck/ui/cocoa/MainApplication.java#L68), I see that the arguments variable is never referenced:

```java
public static void main(String... arguments) {
  // [...]
}
```

Instead it is seems to be implement on the Windows code. Is there any change that the macOS code could be harmonised and gain the ability to load `.duck` files?

Thanks,
Adriano

**Reply by dkocher** [reply]
> I looked at the source code, and unless I'm missing something else, I think the `.duck` file loading is not implemented in macOS. 

This is implemented in https://github.com/iterate-ch/cyberduck/blob/e61b860a13d4b4438a9c8fba82a61cb6efbbdae6/osx/src/main/java/ch/cyberduck/ui/cocoa/controller/MainController.java#L584-L594

**Reply by acataluddi** [reply]
Good morning David,

> This is implemented in 

What you posted is the implementation of the `.duck` file opening. I do not doubt the implementation might exist, as I see the Windows application parsing the command-line.

The point I was trying to make is that, on macOS, it seems that the command-line arguments are not used (so nobody can call and take advantage of `application_openFile()`).

Out of my curiosity, if you launch the following command (pasted by you [here](https://github.com/iterate-ch/cyberduck/discussions/14313#discussioncomment-5207688)), does it work for you meaning that you get *CyberDuck* opening the target connection and actually connecting to it?

```bash
/Applications/Cyberduck.app/Contents/MacOS/Cyberduck ~/Library/Group\ Containers/G69SCX94XU.duck/Library/Application\ Support/duck/Bookmarks/50d3775f-fe75-477c-85e7-7cc801541f76.duck
```

Many thanks,

Adriano C.

**Reply by dkocher** [reply]
The application delegate method `application_openFile` will receive the `.duck` file you pass to the executable from the command line. I just confirmed this to be working as expected.

**Comment by wannesm** [comment]
Thanks for a nice set of products.
Is opening a connection from the command line (or as a custom url scheme) also available for Mountain Duck?

**Reply by tatze96** [reply]
Opening a connection from the command line for Mountain Duck usually works the same as for Cyberduck. However, this currently doesn't work correctly, as the command opens a new instance of Mountain Duck instead of mounting the bookmark within the current instance. We'll investigate this internally and search for a solution.
